backend/inspirehep/records/api/base.py

```diff
@@ -460,6 +461,13 @@ def update(self, data, *args, **kwargs):
             # To be sure that when someone edits redirected record by mistake tries to undelete record as this is not supported for now
             raise CannotUndeleteRedirectedRecord(self.pid_type, data["control_number"])
 
+        if (
+            not force_undelete
+            and self._previous_version.get("deleted")
+            and self.get("deleted") is False
```

Comment:
Here `deleted` key can be None so this check will return False and allow to undelete record while it shouldn't . 
```suggestion
            and self.get("deleted", False) is False
```
OR 
```suggestion
            and not self.get("deleted")
```

Reply:
Ah sorry @drjova  I was too slow ;)